### DISCUSSION TITLE
change email/name author for all commits?

### DISCUSSION THREAD

**Discussion by spaceunifyfifty** [jj-vcs/jj#4513] [Q&A]
I made a bunch of commits without realizing I had an incorrect email address. How can I update all of them to use the newly updated config in my .jj/repo ?

**Comment by emilazy** [comment]
You could try `jj desc --reset-author 'author(exact:"old@email.address")'`, though that would also reset the timestamps.

**Reply by martinvonz** [reply]
Now that #4466 has landed, another option is to run `jj describe --author "My Name <my.name@example.com>"`.

**Reply by mhw** [reply]
A year on, and the more modern option as of v0.33.0 is `jj metaedit --update-author 'author(exact:"wrong@email.address")'`, assuming you've already fixed the jj config author details for the repository. That will rewrite all the mutable commits with the wrong email address. Use `jj log -r 'author(exact:"wrong@email.address")'` to see which commits will be updated before hand.
